micrometer-core/src/main/java/io/micrometer/core/instrument/MeterRegistry.java

```diff
@@ -614,15 +617,29 @@ public Meter remove(Meter meter) {
         return remove(meter.getId());
     }
 
+    /**
+     * Remove a {@link Meter} from this {@link MeterRegistry registry} based on its {@link Id}, optionally
+     * applying this registry's {@link MeterFilter}s to the given {@link Id}.
+     *
+     * @param id the id of the meter to remove
+     * @param applyFilters whether to apply this registry's filters to the id before attempting to remove
+     * @return The removed meter, or null if the meter is not found
+     * @since 1.3.16
+     */
+    @Incubating(since = "1.3.16")
     @Nullable
-    Meter remove(Meter.Id id, boolean applyFilters) {
+    public Meter remove(Meter.Id id, boolean applyFilters) {
```

Comment:
Since this will add public API, I was thinking about it more, and I'm not sure I like this signature, given passing `false` is the same behavior as not passing the parameter at all (existing methods). I'm wondering if it is better to make a different method name. Therein lies the naming bikeshed, though.
* `removeByPreFilterId(Meter.Id preFilterId)`
Something along the lines of this is what I'm leaning towards.

Reply:
Ok, it could be Ok, but in that case, don't forget to modify the "CompositeMeterRegistry.java" file to replace `r -> r.remove(m.getId(), true))` per `r -> r.removeByPreFilterId(m.getId()))`